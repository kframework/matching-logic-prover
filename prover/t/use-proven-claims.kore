// Tests that we can reuse proven claims

symbol f(Int) : Int

axiom a1: \equals(f(1), 1)

claim a2: \equals(f(f(1)), 1)

strategy apply-equation -> a1 at 0 by []
       . apply-equation -> a1 at 0 by []
       . reflexivity


claim a3: \equals(f(f(f(1))), 1)
strategy apply-equation -> a1 at 0 by []
       . apply-equation -> a2 at 0 by [] // <- HERE
       . reflexivity

