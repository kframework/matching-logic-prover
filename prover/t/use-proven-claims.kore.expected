<prover>
  <exit-code>
    0
  </exit-code>
  <goals>
    <goal>
      id: ., parent: .
      <claim>
        \and ( .Patterns )
      </claim>
      <k>
        .
      </k>n<expected>
        .
      </expected>
      <local-context>
        .LocalDeclCellSet
      </local-context>
    <trace>
      .
    </trace>
  </goals>
  <declarations>
    <declaration>
      axiom a1 : \equals ( f ( 1 , .Patterns ) , 1 )
    </declaration> <declaration>
      axiom a2 : \equals ( f ( f ( 1 , .Patterns ) , .Patterns ) , 1 )
    </declaration> <declaration>
      axiom a3 : \equals ( f ( f ( f ( 1 , .Patterns ) , .Patterns ) , .Patterns ) , 1 )
    </declaration> <declaration>
      symbol f ( Int , .Sorts ) : Int
    </declaration>
  </declarations>
</prover>
